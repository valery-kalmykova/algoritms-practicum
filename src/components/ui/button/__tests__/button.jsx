import renderer from 'react-test-renderer';
import { Button } from '../button';
import { render, screen, fireEvent } from '@testing-library/react';

const buttonText = 'Развернуть'
describe('button component tests', () => {

  it('button with text', () => {
    const button = renderer.create(<Button text={buttonText}/>).toJSON()
    expect(button).toMatchSnapshot()
  })

  it('button without text', () => {
    const button = renderer.create(<Button />).toJSON()
    expect(button).toMatchSnapshot()
  })

  it('button is disabled', () => {
    const button = renderer.create(<Button disabled={true} />).toJSON()
    expect(button).toMatchSnapshot()
  })

  it('button is loading', () => {
    const button = renderer.create(<Button isLoader={true} />).toJSON()
    expect(button).toMatchSnapshot()
  })

  it('button has callback onClick', () => {
    const fn = jest.fn();
    render(<Button onClick={fn} text={buttonText} />)
    const button = screen.getByText(buttonText);
    fireEvent.click(button);
    expect(fn).toBeCalledTimes(1);    
  })

})