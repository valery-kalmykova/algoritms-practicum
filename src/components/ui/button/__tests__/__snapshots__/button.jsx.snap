// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component test button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button component test button is loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button component test button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`button component test button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button component tests button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button component tests button is loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`button component tests button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`button component tests button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
